src/storage/FixedConvertingStore.ts

```diff
@@ -0,0 +1,45 @@
+import type { Representation } from '../ldp/representation/Representation';
```

Comment:
So on closer inspection, I was surprised that `RepresentationConvertingStore` only covers `getRepresentation`. But I get why.

I think we should extend `RepresentationConvertingStore` to also cover `addResource` and `setRepresentation`, and perform the steps below.

Then we can either a) only load the needed convertors (which acts as a filter) or b) have a `ConditionalConverter`.

Reply:
I had missed the intro text to this PR, so which is crucial to this point. So let me react:

> For the outgoing data it is based on the user preferences while for incoming it is a fixed list.

Not _necessarily_ though. A fixed list can indeed be an approximation / workable definition, or we can just reuse the same preference-based mechanism. But yeah, point taken.

> We could also have a single converting store class which would then take booleans as input to know which parts it has to support. We still would need 2 instances though so that doesn't seem cleaner.

Yeah, not cleaner indeed, and I don't like the booleans.

But how about changing `RepresentationConvertingStore` to have `in[coming]Convertors` and `out[going]Convertors`? And if you set `convertors`, it just works for both?

Reply:
> Not necessarily though. A fixed list can indeed be an approximation / workable definition, or we can just reuse the same preference-based mechanism.

The problem is that there are no preferences for PUT/POST requests. Unless you mean taking a preferences list as input. That would be no issue since the fixed list gets converted to preferences anyway before sending it to the converter.

> But how about changing RepresentationConvertingStore to have in[coming]Convertors and out[going]Convertors? And if you set convertors, it just works for both?

Two points/"issues" with this:
 * This store needs to appear twice in our "store chain". Once to convert all outgoing data, once to convert all data that gets directed to the sparql store. This means we would need 2 instances of this class since they would have a different source store they continue the request to.
 * It would need constructor parameters for both functionalities. So the constructor would be something like `constructor(source: ResourceStore, inConverter: RepresentationConverter, outConverter: RepresentationConverter, inPreferences: string[])`

But both instances would only use half of that input, which is why I suggested perhaps having an abstract class that contains their common functions.

Reply:
> Unless you mean taking a preferences list as input.

Yes indeed.

> * This store needs to appear twice in our "store chain".

I think that's okay; that's just specific to our chain then. Others might be different.

> * So the constructor would be something like

It would actually be
```
constructor(source: ResourceStore, {
    convertor?: RepresentationConverter,
    inConverter?: RepresentationConverter = convertor,
    outConverter?: RepresentationConverter = convertor,
    inPreferences?: string[] = [],
})
```

Reply:
You can't have type initializers in an object unfortunately so I'm just going to drop the `convertor` parameter from it.

Reply:
No default params? Argh. Or set in constructor. But yeah, no biggie.

Reply:
Also did the constructor, but there you can't combine the optional `?` thingy with initializers, which caused the typing to be `RepresentationConverter | undefined`, which caused it be 3 lines, etc. 😂